File: umbrella-insurance-react-ui/src/models/users/bets/pendingBets/v1/PendingBet.ts
import { Charity } from "../../../../charities/v1/Charity";
import { Game } from "../../../../games/v1/Game";
import { Unit } from "../../../../units/v1/Unit";
import { Bot } from "../../../bots/v1/Bot";
import { PendingBetState } from "../pendingBetStates/v1/PendingBetState";
import { Session } from "../../../sessions/v1/Session";
import { Fee } from "../../../fees/v1/Fee";
import { AccountBalanceTransaction } from "../../../accountBalances/accountBalanceTransactions/v1/AccountBalanceTransaction";
import { BetType } from "../../betTypes/v1/BetType";
import { OrderType } from "../../../../orderTypes/v1/OrderType";

export class PendingBet {
    id?: number;
    session?: Session;
    createdDateTime?: string;
    game?: Game;
    minimumOdds?: number;
    maximumWagerAmount?: number;
    unit?: Unit;//currency
    bot?: Bot;
    isBotBet?: boolean;
    isCharityBet?: boolean;
    charity?: Charity;
    pendingBetState?: PendingBetState;
    fee?: Fee;
    pendingBetName?: string;
    splitPendingBet1?: PendingBet;
    splitPendingBet2?: PendingBet;
    originalPendingBet?: PendingBet;
    accountBalanceEscrowTransaction?: AccountBalanceTransaction;
    accountBalanceCanceledEscrowTransaction?: AccountBalanceTransaction;
    canceledDateTime?: string;
    betType?: BetType;
    orderType?: OrderType;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.pendingBetName) {
                this.pendingBetName = obj.pendingBetName;
            }
            if(obj?.session) {
                this.session = new Session(obj.session);
            }
            if(obj?.createdDateTime) {
                this.createdDateTime = obj.createdDateTime;
            }
            if(obj?.game) {
                this.game = new Game(obj.game);
            }
            if(obj?.minimumOdds) {
                this.minimumOdds = obj.minimumOdds;
            }
            if(obj?.maximumWagerAmount) {
                this.maximumWagerAmount = obj.maximumWagerAmount;
            }
            if(obj?.unit) {
                this.unit = new Unit(obj.unit);
            }
            if(obj?.bot) {
                this.bot = new Bot(obj.bot);
            }
            if(typeof obj?.isBotBet === "boolean") {
                this.isBotBet = obj.isBotBet;
            }
            if(typeof obj?.isCharityBet === "boolean") {
                this.isCharityBet = obj.isCharityBet;
            }
            if(obj?.charity) {
                this.charity = new Charity(obj.charity);
            }
            if(obj?.pendingBetState) {
                this.pendingBetState = new PendingBetState(obj.pendingBetState);
            }
            if(obj?.fee) {
                this.fee = new Fee(obj.fee);
            }
            if(obj?.pendingBetName) {
                this.pendingBetName = obj.pendingBetName;
            }
            if(obj?.splitPendingBet1) {
                this.splitPendingBet1 = new PendingBet(obj.splitPendingBet1);
            }
            if(obj?.splitPendingBet2) {
                this.splitPendingBet2 = new PendingBet(obj.splitPendingBet2);
            }
            if(obj?.originalPendingBet) {
                this.originalPendingBet = new PendingBet(obj.originalPendingBet);
            }
            if(obj?.accountBalanceEscrowTransaction) {
                this.accountBalanceEscrowTransaction = new AccountBalanceTransaction(obj.accountBalanceEscrowTransaction);
            }
            if(obj?.accountBalanceCanceledEscrowTransaction) {
                this.accountBalanceCanceledEscrowTransaction = new AccountBalanceTransaction(obj.accountBalanceCanceledEscrowTransaction);
            }
            if(obj?.canceledDateTime) {
                this.canceledDateTime = obj.canceledDateTime;
            }
            if(obj?.betType) {
                this.betType = new BetType(obj.betType);
            }
            if(obj?.orderType) {
                this.orderType = new OrderType(obj.orderType);
            }
        }
    }
}

File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/PendingBet.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.TimestampDeserializer;
import com.umbrella.insurance.core.serializers.TimestampSerializer;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;
import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "pending_bets", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "pending_bets_unique", columnNames = {"pending_bet_name"})
})
public class PendingBet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "pending_bet_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "session_id")
    private Session session;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "game_id")
    private Game game;

    @NotNull
    @Column(name = "minimum_odds", nullable = false)
    private Double minimumOdds;

    @NotNull
    @Column(name = "maximum_wager_amount", nullable = false)
    private Double maximumWagerAmount;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @NotNull
    @Column(name = "created_date_time", nullable = false)
    private Timestamp createdDateTime;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "unit_id")
    private Unit unit;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "bot_id")
    private Bot bot;

    @Column(name = "is_bot_bet")
    private Boolean isBotBet;

    @Column(name = "is_charity_bet")
    private Boolean isCharityBet;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "charity_id")
    private Charity charity;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "pending_bet_state_id")
    private PendingBetState pendingBetState;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "fee_id")
    private Fee fee;

    @NotNull
    @Column(name = "pending_bet_name", nullable = false, length = Integer.MAX_VALUE)
    private String pendingBetName;

    @OneToMany(mappedBy = "originalPendingBet", cascade = CascadeType.ALL)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    //@JoinColumn(name = "split_pending_bet_id_1")
    private List<PendingBet> splitPendingBetId1;

    @OneToMany(mappedBy = "originalPendingBet", cascade = CascadeType.ALL)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    //@JoinColumn(name = "split_pending_bet_id_2")
    private List<PendingBet> splitPendingBetId2;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "original_pending_bet_id")
    @JsonIgnore
    private PendingBet originalPendingBet;

    @OneToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "account_balance_escrow_transaction_id")
    private AccountBalanceTransaction accountBalanceEscrowTransaction;

    @OneToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "account_balance_canceled_escrow_transaction_id")
    private AccountBalanceTransaction accountBalanceCanceledEscrowTransaction;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @Column(name = "canceled_date_time")
    private Timestamp canceledDateTime;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "bet_type_id")
    private BetType betType;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "order_type_id")
    private OrderType orderType;

}